app/assets/javascripts/school_administrator_dashboard/dashboard_components/absences_dashboard/SchoolwideAbsences.js

```diff
@@ -3,24 +3,48 @@ import PropTypes from 'prop-types';
 
 import DashboardHelpers from '../DashboardHelpers';
 import SchoolwideAttendance from './SchoolwideAttendance';
+import {apiFetchJson} from '../../../helpers/apiFetchJson';
+import measurePageLoad from '../../../helpers/measurePageLoad';
 
 class SchoolwideAbsences extends React.Component {
+  constructor(props) {
+    super(props);
+    this.state = {
+      dashboardStudents: this.props.dashboardStudents,
+      isLoading: false
+    };
+  }
+
+  componentWillMount() {
+    //if the parent component already has the dashboard data, we don't need to duplicate it
+    if(this.props.dashboardStudents.length) return;
+    const url = `/schools/${this.props.schoolId}/absence_dashboard_data`;
+    this.setState({isLoading: true});
 
-  schoolAbsenceEvents() {
-    const studentRecords = this.props.dashboardStudents;
-    return DashboardHelpers.absenceEventsByDay(studentRecords);
+    apiFetchJson(url)
+      .then(response => { this.setState({dashboardStudents: response, isLoading: false}); });
   }
 
   render() {
+    const {dashboardStudents, isLoading} = this.state;
+
+    if (isLoading) {
+      return <p>Loading ...</p>;
+    }
+
     return (
         <SchoolwideAttendance
-          schoolAbsenceEvents = {this.schoolAbsenceEvents()}
-          dashboardStudents = {this.props.dashboardStudents}/>);
+          schoolAbsenceEvents = {DashboardHelpers.absenceEventsByDay(dashboardStudents)}
+          dashboardStudents = {dashboardStudents}/>);
   }
 }
 
 SchoolwideAbsences.propTypes = {
-  dashboardStudents: PropTypes.array.isRequired
+  dashboardStudents: PropTypes.array.isRequired,
+  schoolId: PropTypes.number
 };
 
+measurePageLoad(info => console.log(JSON.stringify(info, null, 2))); // eslint-disable-line no-console
```

Comment:
This code looks like it will run when this file is imported, not when this component is rendered.  Should this be moved to `App#componentDidMount` to run on startup for now?  Same for the other usage.